File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/ApuracaoDeJurosJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapterDataSource;
import java.math.BigDecimal;
import java.util.Date;

public abstract class ApuracaoDeJurosJRAdapter
extends JRAdapter {
    public abstract JRAdapterDataSource<ApuracaoDeJurosAdapter> getOcorrenciaDeJuros();

    public abstract BigDecimal getTotalDeJuros();

    public abstract class ApuracaoDeJurosAdapter
    extends JRAdapter {
        public abstract Date getCompetencia();

        public abstract Date getDataInicial();

        public abstract BigDecimal getValor();

        public abstract BigDecimal getContribuicaoSocial();

        public abstract BigDecimal getPrevidenciaPrivada();

        public abstract BigDecimal getCapital();

        public abstract BigDecimal getTaxa();

        public abstract BigDecimal getJuros();
    }
}

File: src/lib/pjecalc/core/dominio/calculo/relatorio/apuracao-de-juros-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — ApuracaoDeJurosJRAdapter (abstract)
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.ApuracaoDeJurosJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../relatorio/ApuracaoDeJurosJRAdapter.java
 *
 * Demonstrativo mês a mês da apuração de juros de mora (capital + juros).
 */
import type Decimal from 'decimal.js';
import { JRAdapter, JRAdapterDataSource } from './jr-adapter';

export abstract class ApuracaoDeJurosJRAdapter extends JRAdapter {
  abstract getOcorrencias(): JRAdapterDataSource<ApuracaoDeJurosOcorrenciaAdapter>;
  abstract getTotalCapital(): Decimal;
  abstract getTotalJuros(): Decimal;
  abstract getTotalGeral(): Decimal;
}

export abstract class ApuracaoDeJurosOcorrenciaAdapter extends JRAdapter {
  abstract getCompetencia(): Date;
  abstract getDataInicial(): Date;
  abstract getValorCorrigido(): Decimal;
  abstract getContribuicaoSocial(): Decimal;
  abstract getPrevidenciaPrivada(): Decimal;
  abstract getCapital(): Decimal;
  abstract getTaxaDeJuros(): Decimal;
  abstract getJuros(): Decimal;
  abstract getTotal(): Decimal;
}
